src/inc/HostSignals.hpp

```diff
@@ -0,0 +1,26 @@
+// These values match the enumeration values of `ControlType` for the `ConsoleControl` class
+// but are defined here similarly to not pollute other projects.zs
+// They don't *have* to be the same values, but matching them seemed to make sense.
+#define HOST_SIGNAL_NOTIFY_APP 1u
+struct HOST_SIGNAL_NOTIFY_APP_DATA
+{
+    DWORD cbSize;
+    DWORD dwProcessId;
+};
+
+#define HOST_SIGNAL_SET_FOREGROUND 5u
+struct HOST_SIGNAL_SET_FOREGROUND_DATA
+{
+    DWORD cbSize;
+    ULONG dwProcessId;
+    BOOL fForeground;
+};
+
+#define HOST_SIGNAL_END_TASK 7u
+struct HOST_SIGNAL_END_TASK_DATA
+{
+    DWORD cbSize;
+    ULONG dwProcessId;
+    DWORD dwEventType;
+    ULONG ulCtrlFlags;
+};

```

Comment:
for "wire types", it might be better to prefer the known sized types from stdint, `uint8_t` `uint32_t` `uint16_t` etc.

Also: we don't have to be weird -- we can just use enums like civilized folks instead of defines like.. define-users :)

Reply:
especially because enum classes make it possible to enforce the enum type size on the wire